### PR TITLE
Missed refs to Dogwood docs

### PR DESCRIPTION
**Pull request by lamagnifica**

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
## [DOC-2643](https://openedx.atlassian.net/browse/DOC-2643)

Sure enough, I missed the cover page in the Open edX Release Notes guide which lists the released versions with xrefs. All wrong. 

This guide is NOT versioned, and is only delivered as latest, so this PR is off of master.

### Date Needed

16 Feb if possible

### Reviewers
- [x] Subject matter expert: @nedbat 
- [x] Doc team review (sanity check): @pdesjardins @catong @srpearce 

### Testing
- [x] Ran ./run_tests.sh without warnings or errors

### Post-review
- [x] Squash commits

**Comment by nedbat** [comment]
:+1:

File: en_us/open_edx_release_notes/source/os_documentation.rst [hunk 1]
```diff
@@ -5,44 +5,45 @@ Open edX Documentation
...
-of the following documents for Open edX users.
+In addition to the documentation for the most recent release of Open edX, edX
+also maintains the latest version of the following documents for Open edX
+users.
 
-.. note:: 
+.. note::
   The latest versions of these documents might contain changes related
-  to functionality that is not present in the Cypress release.
+  to functionality that is not present in the most recent release.
```
Comment (srpearce) [inline en_us/open_edx_release_notes/source/os_documentation.rst:29]: Maybe change to "most recent named release"?

**Comment by srpearce** [comment]
One minor comment, otherwise :+1:

**Comment by lamagnifica** [comment]
@srpearce there is no comment in the files, perhaps try again?

File: en_us/open_edx_release_notes/source/os_documentation.rst (hunk 1 above)
Comment (lamagnifica) [inline en_us/open_edx_release_notes/source/os_documentation.rst:29]: we took out all references to "named" releases, and replaced them with just release or {name of release} release after Cypress went out last year.

Comment (nedbat) [inline en_us/open_edx_release_notes/source/os_documentation.rst:29]: Or, "most recent Open edX release."

**Pull request merged**